encoding/rs/encode.go

```diff
@@ -127,3 +144,38 @@ func (g *Encoder) ExtendPolyEval(coeffs []fr.Element) ([]fr.Element, []fr.Elemen
 
 	return evals, pdCoeffs, nil
 }
+
+type JobRequest struct {
+	Index      uint64
+	FrameIndex uint64
+}
+
+func (g *Encoder) interpolyWorker(
+	polyEvals []fr.Element,
+	jobChan <-chan JobRequest,
+	results chan<- error,
+	frames []Frame,
+) {
+
+	for jr := range jobChan {
+		i := jr.Index
+		k := jr.FrameIndex
+		j := rb.ReverseBitsLimited(uint32(g.NumChunks), uint32(i))
+		ys := polyEvals[g.ChunkLength*i : g.ChunkLength*(i+1)]
+		err := rb.ReverseBitOrderFr(ys)
+		if err != nil {
+			results <- err
+			continue
+		}
+		coeffs, err := g.GetInterpolationPolyCoeff(ys, uint32(j))
+		if err != nil {
+			results <- err
+			continue
+		}
+
+		frames[k].Coeffs = coeffs
+	}
+
+	results <- nil
```

Comment:
should this be inside the for loop? It'll be then one result per JobRequest

Reply:
there are "numWorker" of threads, and "NumChunks" of jobs. If it is moved inside the for loop, the worker is terminated